SAST Report on Access to field 'pNext' results in a dereference of a null pointer

## Maintainer Report

Mon Jan  5 02:11:34 AM UTC 2026

Issue: Access to field 'pNext' results in a dereference of a null pointer (loaded from variable 'last')
SAST Log: https://github.com/InternationalColorConsortium/iccDEV/actions/runs/20702517319/job/59427126296#step:7:298


## SAST log

```
  /home/runner/work/iccDEV/iccDEV/IccProfLib/IccCmm.cpp:11186:17: warning: Access to field 'pNext' results in a dereference of a null pointer (loaded from variable 'last') [core.NullDereference]
   11186 |     last->pNext = m_pFirst;
         |     ~~~~        ^
  1 warning generated.
  
```

## Request

Please consider refactoring and/or adding comments to the code if this is the intended behavior when time permits. If this is the intended behavior please Label as, OOS, wontfix and Close.

Thank You

--- a/IccProfLib/IccCmm.cpp
+++ b/IccProfLib/IccCmm.cpp
@@ -84,6 +84,7 @@
 #include "IccSparseMatrix.h"
 #include "IccEncoding.h"
 #include "IccMatrixMath.h"
+#include <cassert>
 
 #ifdef USEICCDEVNAMESPACE
 namespace iccDEV {
@@ -11183,6 +11184,10 @@ bool CIccMruCache<T>::Apply(T *DstPixel, const T *SrcPixel)
   else {  //Reuse oldest value and put it at the front of the list
     if (prev)
       prev->pNext = NULL;
+
+    // Static analysis gets a false positive here, because it doesn't understand the relation between count and pointers in the list
+    // assert avoids the static analysis warning, and should never fire
+    assert( m_pFirst != NULL );
     last->pNext = m_pFirst;
 
     m_pFirst = last;

